apps/web-connect/src/features/checkout/Checkout.tsx

```diff
@@ -40,8 +42,9 @@ export function Checkout() {
     useEffect(() => {
         if (prefilledPromoCode) {
             setPromoCode(prefilledPromoCode);
+            handleApplyPromoCode();
         }
-    }, [prefilledPromoCode, setPromoCode]);
+    }, [promoCode, setPromoCode]);
```

Comment:
This will keep rerendering for infinity.
The useEffect should not have any dependencies, this should only run once on component mount.

Then, when I tried this, I could not get the promocode to auto fill, am I missing something ?
Can you verify that you tested this locally ?

Reply:
Yes I tested this locally and confirmed it worked - this was the only way I could get it to render at all

Reply:
@CryptITAustria @craigbranscom are we sure this will keep re-rendering for infinity?

If that were the case, wouldn't it have already been happening with the prefilledPromoCode variable that was previously there?